thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/detection/alert/scheme/DetectionEmailAlerter.java

```diff
@@ -66,88 +49,17 @@
 public class DetectionEmailAlerter extends DetectionAlertScheme {
   private static final Logger LOG = LoggerFactory.getLogger(DetectionEmailAlerter.class);
 
-  private static final Comparator<AnomalyResult> COMPARATOR_DESC = new Comparator<AnomalyResult>() {
-    @Override
-    public int compare(AnomalyResult o1, AnomalyResult o2) {
-      return -1 * Long.compare(o1.getStartTime(), o2.getStartTime());
-    }
-  };
-
-  private static final DAORegistry DAO_REGISTRY = DAORegistry.getInstance();
-
+  private static final Comparator<AnomalyResult> COMPARATOR_DESC =
+      (o1, o2) -> -1 * Long.compare(o1.getStartTime(), o2.getStartTime());
   private static final String DEFAULT_EMAIL_FORMATTER_TYPE = "MultipleAnomaliesEmailContentFormatter";
 
-  private DetectionAlertConfigManager alertConfigDAO;
-  private final DetectionAlertConfigDTO config;
-  private final DetectionAlertFilterResult result;
   private ThirdEyeAnomalyConfiguration thirdeyeConfig;
-  private CurrentAndBaselineLoader currentAndBaselineLoader;
 
   public DetectionEmailAlerter(DetectionAlertConfigDTO config, TaskContext taskContext,
```

Comment:
Maybe instead of passing the whole taskContext, only pass in part of the ThirdEyeAnomalyConfiguration that is needed in this alerter.

Reply:
Yes, I initially considered to just pass the alerter related configuration, but apparently, there are a lot of parameters that are required especially to populate the emailConfig like the dashboard host,  failureEmails, functionConfig, alertFilter etc.

Reply:
I'm not a fan of this either. That explicitly couples the alerter to the task execution framework. Since you only keep the ThirdEye config - please pass it explicitly.

Reply:
Ah! I think I misunderstood Jihao's point. +1 to passing only the thirdeye config but I cannot pass part of the thirdeyeconfig.